### ISSUE TITLE
[Notifications] Ability to control types of enabled desktop notifications

### ISSUE THREAD

**Issue by tomneedham** [owncloud/client#6443]
<img width="291" alt="screen shot 2018-04-10 at 12 01 21" src="https://user-images.githubusercontent.com/660805/38553376-003315ac-3cb7-11e8-8423-2965842252ba.png">

This checkbox decides whether I get native desktop notifications - which is cool. But, in my use case, I really don't want to see every action I perform as a notification on my desktop. If its a received share, and another user changes something, maybe, but actions I perform on my own set of data I dont want to appear as notifications - so basically I have always had this option disabled.

However - getting ownCloud notifications as desktop notifications is awesome, like acept/reject shares, new updates, annoucement etc is AWESOME! I would like to be able to distinguish between them so I can have those enabled, but hide the normal sync activity notifications.

**Comment by tomneedham** [comment]
Just my thoughts / ideas :)

**Comment by SamuAlfageme** [comment]
@tomneedham Hm. I think I get it. We should distinguish between two types of notifications:

- Related to server activity -> appear on the "Activity" tab
- Coming from the sync process itself -> link to "Sync Protocol / Not Synced" tabs with different events: 
    - Conflicts
    - Errors

Both types get a native notification. You wanna see the ones from the server and **filter out** the frequency of the latter (e.g. get conflicts and errors but no notification on sync success - right?).

> I really don't want to see every action I perform as a notification on my desktop

How's your user experience with this? Do you have examples of which notifications would you like to see less frequently?

And about the OCS' notifications - API does not differentiate types, they're all inside a `data` array:

```json
{
    "ocs": {
        "meta": {
            "status": "ok",
            "statuscode": 200,
            "message": null,
            "totalitems": "",
            "itemsperpage": ""
        },
        "data": [
            {
                "notification_id": 1,
                "app": "market",
                "user": "admin",
                "datetime": "2018-04-10T08:11:02+00:00",
                "object_type": "templateeditor",
                "object_id": "0.3",
                "subject": "Update for Mail Template Editor to version 0.3 is available.",
                "message": "",
                "link": "http://localhost/index.php/apps/market/#/app/templateeditor",
                "actions": [],
                "icon": ""
            }
        ]
    }
}
```

However, we could filter out (whitelist) which `app`s' notifications to display on the activity tab based on the user selection. e.g. Since marketplace was released, being an admin gets you lots of "_update the app X_" desktop notifications.

**Comment by ckamm** [comment]
I also think this makes sense. It's purely informational can-be-ignored notifications versus action-needed or critical notifications.

**Comment by tomneedham** [comment]
@SamuAlfageme correct. There should be two checkboxes, not sure on naming but as you said:

 - Server notifications (those shown in the bell)
 - Sync protocol notifications
